src/lib/components/i18n/index.svelte

```diff
@@ -0,0 +1,15 @@
+<script lang="ts">
+  import { interpolate } from '$lib/utilities/i18n.utilities';
+
+  export let html = false;
+  export let key: string;
+  export let values: Record<string, string> | undefined = undefined
+
+  let message = interpolate(key, values);
+</script>
+
+{#if html}
+  <slot>{@html message}</slot>
```

Comment:
This is the main reason for this component. The `@html` is the same as `dangerouslySetInnerHtml` in React. Since it's a liability, I wanted to do it in one place, and only after I know the contents have been cleaned by the `interpolate` function.

Reply:
Ah so this is how we might let for example the step.getHeader() text render HTML - once it has been sanitized? I think I get this now

Reply:
This is definitely possible, but it's an open question about how to handle localization from a system perspective. I'd, personally, recommend all content come from the localization files within the project as this lines up better with how large company's handle this organizationally. That way, you have one source of truth that can be updated and controlled by the same team.

Now, there are instances where this isn't 100% possible, and some localized content does come from AM/IDM. This is not ideal, but it's just something we have to deal with. When it comes to choices, like page node headers and descriptions, I'd recommend they use a single value in AM as a key to lookup the content from within the app, even though it _could_ be localized in AM.

I say that because most big companies have a localization team that's responsible for maintaining the locale files for the user facing apps. UI engineers rarely worry about these files, and they just provide the US English strings provided by the design or content team. The localization team then takes the provided content and adds the new keys and translated strings to the supported locale files. They are not likely to have access or want to rummage through AM for translated strings.

Does that make sense? This topic gets complicated and deep very quickly.